### PR TITLE
CONCF-189 move social icons rendering to client side

### PR DESCRIPTION
**Pull request by bognix**

### CODE REVIEW

**Comment by bognix** [comment]
ping @vforge @rogatty

File: extensions/wikia/PageShare/scripts/PageShare.js [hunk 1]
```diff
@@ -7,6 +7,34 @@ require(['wikia.window', 'wikia.tracker', 'jquery'], function(win, tracker, $) {
...
+			method: 'getShareIcons',
+			data: {
+				browserLang: browserLangShort,
+				useLang: useLang
+			},
+			callback: pasteShareIcons
+		})
+	}
+
+	function pasteShareIcons(data) {
```
Comment (rogatty) [inline extensions/wikia/PageShare/scripts/PageShare.js]: More like `append` or `render`. It's not clipboard ;)

File: extensions/wikia/PageShare/PageShareHelper.class.php [hunk 2]
```diff
@@ -22,25 +22,30 @@ public static function getIcon( $service ) {
...
+	public static function getLangForPageShare( $browserLang, $useLang ) {
+		global $wgLang, $wgUser;
 
-		if ( $wgUser->isAnon() ) {
-			if ( !empty( $_SERVER[ 'HTTP_ACCEPT_LANGUAGE' ] ) ) {
-				$lang = substr( $_SERVER['HTTP_ACCEPT_LANGUAGE'], 0, 2 );
+		if ( !empty ( $useLang ) ) {
+			return $useLang;
+		} else {
+			if ( $wgUser->isAnon() ) {
```
Comment (rogatty) [inline extensions/wikia/PageShare/PageShareHelper.class.php]: You can decrease nesting with `else if`

File: extensions/wikia/PageShare/scripts/PageShare.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		trackFunc({label: service.data('share-service')});
+
+		win.open(url, title, 'width=' + w + ',height=' + h);
+	}
+
+	function appendShareIcons(data) {
+		var $container = $('#PageShareContainer');
+		if (data.socialIcons) {
+			$container.html(data.socialIcons);
+			$('#PageShareToolbar').on('click', '.page-share a', shareLinkClick);
```
Comment (rogatty) [inline extensions/wikia/PageShare/scripts/PageShare.js]: Would it work with `$container.on(...)`?

File: extensions/wikia/PageShare/tests/PageShareHelperTest.php [hunk 4]
```diff
@@ -128,29 +111,27 @@ public function testGetLangForPageShare(
...
 		return [
-			[ null, null, true, true, 'pl,pl;q=0.8', 'pl' ],
-			[ null, null, true, false, 'pt-br,pt;q=0.8', 'en' ],
-			[ null, null, true, true, false, 'en' ],
-			[ 'bar', null, true, true, 'pt-br,pt;q=0.8', 'bar' ],
-			[ null, 'foo', false, false, null, 'foo' ],
-			[ 'bar', 'foo', false, false, null, 'bar' ]
+			[ null, null, true, 'pl', 'pl' ],
+			[ null, null, true, false, 'en' ],
+			[ 'bar', null, true, 'pt-br,pt;q=0.8', 'bar' ],
```
Comment (rogatty) [inline extensions/wikia/PageShare/tests/PageShareHelperTest.php]: Does it still make sense to test PHP code against `'pt-br,pt;q=0.8'` if we're trimming it in the JS now (https://github.com/Wikia/app/pull/6893/files#diff-e5bdf00a54479fd4a44d3ec365b951a9R44)?

Comment (bognix) [inline extensions/wikia/PageShare/tests/PageShareHelperTest.php]: The main goal of this test is to check if "bar" is overwriting value of "pt..." so the value of browser lang argument is irrelevant. I can change it to "foo" if it increases readability.

Comment (rogatty) [inline extensions/wikia/PageShare/tests/PageShareHelperTest.php]: Let's increase readability by creating test data similar to realword use cases :dancers:

**Comment by vforge** [comment]
LGTM

**Pull request merged**